airflow/sensors/external_task_sensor.py

```diff
@@ -70,9 +76,24 @@ def __init__(self,
         self.execution_date_fn = execution_date_fn
         self.external_dag_id = external_dag_id
         self.external_task_id = external_task_id
+        self.check_existence = check_existence
 
     @provide_session
     def poke(self, context, session=None):
+        TI = TaskInstance
+
+        if self.check_existence:
+            existence = session.query(TI).filter(
+                TI.dag_id == self.external_dag_id,
+                TI.task_id == self.external_task_id,
+            ).count()
+            session.commit()
+            if existence == 0:
+                raise AirflowException('The external task "' +
```

Comment:
why stop waiting if the external task not exist? Shouldn't the right behavior to continue waiting for the task until task exist and finish(or timeout?). Sometimes the external task may not start right on time(e.g scheduler reason etc) and we still want downstream task wait for external task to finish first to make sure result align.

Reply:
There may be a few cases:
- The external DAG ID specified is wrong (due to reasons like typo);
- The external task specified doesn't exist in the corresponding DAG (similar reason).
- ...

Starting on time or not doesn't matter much here, since only DAG ID& task ID are used here for querying.

I consider this feature as a "guard" to prevent from entering wrong DAG ID and task ID. Without guard, it may be hard to find this type of errors (the sensor will keep waiting and eventually leaves an impression that it fails only because the external task was not executed/finished yet).

Given the default value of this new argument is `False`, it will not change the current behavior.

Reply:
But I do agree this feature may be too specific and only applicable to limited use cases (like some I'm dealing with).

Please share your thoughts, and feel free to close this PR should you find pro < con. Thanks.